documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -28,96 +28,100 @@ We request OpenAPI(Swagger) spec authoring be assigned to engineers who have an
-| M2017	| PutRequestResponseValidation | A PUT operation request body schema should be the same as its 200 response schema, to allow reusing the same entity between GET and PUT. If the schema of the PUT request body is a superset of the GET response body, make sure you have a PATCH operation to make the resource updatable. Operation: '{0}' Request Model: '{1}' Response Model: '{2}' | Warning |
-| [R1009](#R1009) | [DeleteInOperationName](#R1009) | 'DELETE' operation '{0}' should use method name 'Delete'. Note: If you have already shipped an SDK on top of this spec, fixing this warning may introduce a breaking change. | Warning |
-| [R1005](#R1005) | [GetInOperationName](#R1005) | 'GET' operation '{0}' should use method name 'Get' or Method name start with 'List'. Note: If you have already shipped an SDK on top of this spec, fixing this warning may introduce a breaking change. | Warning |
-| [R1003](#R1003) | [ListInOperationName](#R1003) | Since operation '{0}' response has model definition '{1}', it should be of the form "*_list*" | Warning |
-| [R1006](#R1006)	| [PutInOperationName](#R1006) | 'PUT' operation '{0}' should use method name 'Create'. Note: If you have already shipped an SDK on top of this spec, fixing this warning may introduce a breaking change. | Warning |
-| [R1007](#R1007) | [PatchInOperationName](#R1007) | 'PATCH' operation '{0}' should use method name 'Update'. Note: If you have already shipped an SDK on top of this spec, fixing this warning may introduce a breaking change. | Warning |
-| [R1011](#R1011) | [HttpsSupportedScheme](#R1011) | 'Azure Resource Management only supports HTTPS scheme. | Warning |
-| [R2065](#R2065) | [LicenseHeaderMustNotBeSpecified](#R2065) | License header must not be specified inside x-ms-code-generation-settings of OpenAPI document. | Warning |
-| [R2018](#R2018) | [XmsEnumValidation](#R2018) | The enum types should have x-ms-enum type extension set with appropriate options. Property name: {0}. | Warning |
-| [R2063](#R2063) | [OperationIdNounConflictingModelNames](#R2063) | OperationId has a noun that conflicts with one of the model names in definitions section. The model name will be disambiguated to '{0}Model'. Consider using the plural form of '{1}' to avoid this. Note: If you have already shipped an SDK on top of this spec, fixing this warning may introduce a breaking change. | Warning |
+| Id | Rule Name |
+| --- | --- |
+| M4000 | ModelTypeIncomplete |
+| [R4000](#R4000-2) | [ParameterDescriptionRequired](#R4000-2) |
+| [R4000](#R4000-3) | [DescriptiveDescriptionRequired](#R4000-3) |
+| [R4000](#R4000-4) | [DescriptionAndTitleMissing](#R4000-4) |
+| [R4000](#R4000-5) | [OperationDescriptionOrSummaryRequired](#R4000-5)  |
+| [R2001](#R2001)	| [AvoidNestedProperties](#R2001) |
+| [R4002](#R4002)	| [LocationMustHaveXmsMutability](#R4002) |
+| [R2066](#R2066)	| [PostOperationIdContainsUrlVerb](#R2066) |
```

Comment:
The id needs to change in Typescript. I will create a seperate PR for changing the Id